lib/msal-common/src/response/ResponseHandler.ts

```diff
@@ -599,14 +600,24 @@ export class ResponseHandler {
                 serverTokenResponse?.spa_accountid;
         }
 
+        let accountInfo: AccountInfo | null = null;
+        if (cacheRecord.account && idTokenClaims) {
+            const fullAccountEntity: AccountEntity = Object.assign(
+                new AccountEntity(),
+                {
+                    ...cacheRecord.account,
+                    idTokenClaims: idTokenClaims,
+                }
+            );
+            accountInfo = fullAccountEntity.getAccountInfo();
+        }
```

Comment:
Does this work instead?
```suggestion
            accountInfo = { ...cacheRecord.account.getAccountInfo(), idTokenClaims };
```

Reply:
Believe I saw this in one other place as well

Reply:
Ideally the AccountEntity doesn't have idTokenClaims defined on it at all if they won't be cached.